src/main/java/com/amazon/ion/impl/bin/IonManagedWriter_1_1.kt

```diff
@@ -278,8 +285,10 @@ internal class IonManagedWriter_1_1(
             if (sid == UNKNOWN_SYMBOL_ID) {
                 // No (known) SID either.
                 throw UnknownSymbolException("Cannot write a symbol token with unknown text and unknown SID.")
-            } else {
+            } else if (sidTransformer == null) {
                 rawWriter.write(kind, sid)
+            } else {
+                rawWriter.write(kind, sidTransformer!!.transform(sid))
```

Comment:
This is more of a FYI than a request for a change.
```suggestion
            } else {
                rawWriter.write(kind, sidTransformer?.transform(sid) ?: sid)
```

Reply:
Done, thanks